cms/static/js/hera/components/EndSurvey.jsx

```diff
@@ -1,144 +1,267 @@
-                                                <input className="questions__list__input" type="radio" />
-                                                <div className="questions__list__text">
-                                                    <input className="questions__list__text-hint" placeholder="Question text" type="text" />
-                                                </div>
-                                            </label>
-                                            <div className="end-survey__radio-buttons">
-                                                <button className="end-survey__radio-btn is-add" type="button">
-                                                    <i className="fa fa-plus-square" aria-hidden="true" />
-                                                </button>
-                                                <button className="end-survey__radio-btn is-remove" type="button">
-                                                    <i className="fa fa-trash-o" aria-hidden="true" />
-                                                </button>
-                                            </div>
-                                        </div>
-                                    </div>
-                                </div>
-                            </div>
+                            {!!(questions.length) && questions.map((item, idx) => {
+                                return (
+                                        <SurveyQuestion
+                                            key={idx}
+                                            index={idx}
+                                            removeHandler={this.removeEndSurveyQuestion}
+                                            changeHandler={this.endSurveyQuestionChanged}
+                                            {...item}/>
+                                    )
+                            })}
 
                             <div className="end-survey__row-buttons">
-                                <button className="end-survey__row-btn is-remove" type="button">
-                                    <i className="fa fa-trash-o" aria-hidden="true" />
-                                    Remove row
-                                </button>
-                                <button className="end-survey__row-btn is-add" type="button">
-                                    <i className="fa fa-plus-circle" aria-hidden="true" />
-                                    Add row with items
+                                <button className="end-survey__row-btn is-add" type="button" onClick={this.addEndSurveyQuestion}>
+                                    <i className="fa fa-plus-circle" aria-hidden="true"/>
+                                    Add a question
                                 </button>
                             </div>
                         </div>
+                        <div className="end-survey__row">
```

Comment:
Why do you add this question separately here? Isn't it better just to add a record to `questions` storage so it can be rendered by https://github.com/raccoongang/edx-platform/pull/1797/files#diff-58442bc132f90fb961ffdfcb70d39bceR217

Reply:
I separated this one because this is another kind of question and it's required, it can not be deleted. the questions above are the Survey Questions which can be added/edited/removed and the last one is the Confidence Question as specification requires.